### PR TITLE
Fixed dangerous default argument

### LINKED ISSUE #4227: (PYL-W0102) Dangerous default argument
## Description
Mutable Default Arguments: https://docs.python-guide.org/writing/gotchas/

Do not use a mutable like `list` or `dictionary` as a default value to an argument. Python’s default arguments are evaluated once when the function is defined. Using a mutable default argument and mutating it will mutate that object for all future calls to the function as well.

## Occurrences
There are 23 occurrences of this issue in the repository.

See all occurrences on DeepSource &rarr; [deepsource.io/gh/HarshCasper/openlibrary-1/issue/PYL-W0102/occurrences/](https://deepsource.io/gh/HarshCasper/openlibrary-1/issue/PYL-W0102/occurrences/)

### PR DESCRIPTION
**Pull request by Yashs911**
Related to #4227 

Create a new object each time the function is called, by using a default arg to signal that no argument was provided.
### Stakeholders

@cclauss

### CODE REVIEW

File: openlibrary/core/cache.py [hunk 1]
```diff
@@ -160,7 +160,9 @@ def join_threads(self):
         for name, thread in self.active_threads.items():
             thread.join()
 
-    def encode_args(self, args, kw={}):
+    def encode_args(self, args, kw=None):
+        if kw is None:
+            kw = kw or {}
```
Comment (cclauss) [inline openlibrary/core/cache.py:165]:
```suggestion
        kw = kw or {}
```
Repeat with all other instances.

File: openlibrary/plugins/upstream/adapter.py [hunk 2]
```diff
@@ -105,7 +105,10 @@ def process_error(self):
                 d['key'] = unconvert_key(d['key'])
             self.output = simplejson.dumps(d)
 
-def convert_key(key, mapping=convertions):
+
```
Comment (cclauss) [inline openlibrary/plugins/upstream/adapter.py:108]: Please undo the changes to this file.

File: openlibrary/tests/solr/test_update_work.py [hunk 3]
```diff
@@ -75,7 +75,9 @@ class FakeDataProvider(DataProvider):
     docs = []
     docs_by_key = {}
 
-    def __init__(self, docs=list()):
+    def __init__(self, docs=None):
+        if docs is None:
+            docs = docs or list()
```
Comment (cclauss) [inline openlibrary/tests/solr/test_update_work.py:80]:
```suggestion
        docs = docs or []
```

File: openlibrary/plugins/upstream/adapter.py (hunk 2 above)
Comment (Yashs911) [reply openlibrary/plugins/upstream/adapter.py:108]: Okay

Comment (cclauss) [reply openlibrary/plugins/upstream/adapter.py:108]: Please remove lines 108 and 125 so there are no changes to this file.

**Review by cclauss — APPROVED** [review]
Nice!  Thanks for doing this!

File: openlibrary/plugins/openlibrary/tests/test_ratingsapi.py [hunk 4]
```diff
@@ -26,7 +26,8 @@ def __init__(self, config):
         self.opener.add_handler(
             urllib.request.HTTPCookieProcessor(self.cookiejar))
```
Comment (elTiempoVuela) [inline openlibrary/plugins/openlibrary/tests/test_ratingsapi.py:27]: HTTPCookieProcessor
